liblwgeom/lwgeom_median.c

```diff
@@ -28,43 +28,50 @@
 #include "lwgeom_log.h"
 
 static void
-calc_distances_3d(const POINT3D* curr, const POINT3D* points, uint32_t npoints, double* distances)
+calc_weighted_distances_3d(const POINT4D* curr, const POINT4D* points, size_t npoints, double* distances)
 {
-	uint32_t i;
```

Comment:
PostGIS point array size is limited to 32-bit. (I'm not sure why all counters were changed to `size_t` ?)

Reply:
Add comment to clarify that `m` ordinate is taken to be a weight

Reply:
Accessing 32-bit variables in 64-bit environments is slower in my experience, especially in loops. To get that right across architectures maps.me uses internal convention to use `size_t` for counters.

Documentation change should be there, at least `Enhanced:` entry.